File: AlHadoop/src/main/java/com/al/util/StringUtil.java
package com.al.util;

import java.io.UnsupportedEncodingException;

public class StringUtil {
	/**
	 * 标题和Url截断
	 * @param str
	 * @param len
	 * @param encoding
	 * @return
	 */
	public static String limitString(String str, int len, String encoding) {
		String s = str;
		try {
			byte[] yy = s.getBytes(encoding);
			while(s != null && yy.length > len) {
				int sublen = getEncodingLen(yy.length, s.length(), len);
				if(sublen < 0) {
					sublen = s.length() / 2;
				}
				s = s.substring(0, sublen);
				yy = s.getBytes(encoding);
			}
			return s;
		} catch (UnsupportedEncodingException e) {
			e.printStackTrace();
		}
		return s;
	}

	private static int getEncodingLen(int bytelen, int strlen, int maxlen) {
		return strlen-(bytelen - maxlen);
	}

	/**
	 * 标题清理过滤
	 * @param title
	 * @return
	 */
	public static String clearTitleAll(String title) {
		String result = clearTitle(title, "-");
		result = clearTitle(result, "_");
		return result;
	}

	public static String clearTitle(String title, String symbol) {
		if(title == null) {
			return "";
		}
		int index = title.lastIndexOf(symbol);
		return index < 5 ? title : clearTitle(title.substring(0, index), symbol);
	}
}

File: AlSpark/src/main/scala/com/al/util/StringUtil.scala
package com.al.util

object StringUtil {

	/**
		* 标题和Url截断
		* @param str
		* @param len
		* @param encoding
		* @return
		*/
	def limitString(str: String , len: Int , encoding: String): String = {
		  var s = str
			var yy: Array[Byte] = s.getBytes(encoding);
			while(s != null && yy.length > len) {
				var sublen: Int  = getEncodingLen(yy.length, s.length(), len);
				if(sublen < 0) {
					sublen = s.length() / 2;
				}
				s = s.substring(0, sublen);
				yy = s.getBytes(encoding);
			}
			return s;
	}
	
	def getEncodingLen(bytelen: Int, strlen: Int , maxlen: Int): Int = {
		return strlen - (bytelen - maxlen);
	}

	/**
		* 标题清理过滤
		* @param title
		* @return
		*/
	def clearTitleAll(title: String): String = {
		var result: String = clearTitle(title, "-")
		result = clearTitle(result, "_")
		return result
	}

	def clearTitle(title: String, symbol: String): String = {
		if(title == null) {
			return ""
		}
		val index: Int = title.lastIndexOf(symbol)
		if(index < 5) {
			return title
		} else {
			return clearTitle(title.substring(0, index), symbol)
		}
	}
}